test/controllers/trips_controller_test.rb

```diff
@@ -1,23 +1,99 @@
 require "test_helper"
 
 describe TripsController do
+  let (:trip) {
+    Trip.create
+  }
+
+  describe "index" do 
+    it "can get a list of all trips" do 
+
+      get trips_path 
+
+      must_respond_with :success 
+    end 
+  end 
+
   describe "show" do
-    # Your tests go here
+
+    it "can get trip" do 
+      passenger = Passenger.create(name: "Nina", phone_num: "12345")
+      driver = Driver.create(name: "Anna", vin: "qwert123", available: true)
+      new_trip = Trip.create(date: "today", rating: 0, cost: Trip.trip_cost, driver_id: driver.id, passenger_id: passenger.id)
+
+     
+      get trip_path(new_trip.id)
+      must_respond_with :success
+    end 
+
+    it "will redirect for invalid trip" do 
+      get trip_path(-1)
+      must_respond_with :redirect
+
+    end 
   end
 
   describe "create" do
-    # Your tests go here
+    it "can request new trip" do 
+
+      passenger = Passenger.create(name: "Nina", phone_num: "12345")
+      driver = Driver.create(name: "Anna", vin: "qwert123", available: true)
+
+      trip_hash = {
+        trip: {
+          date: "today",
+          rating: 0,
+          cost: Trip.trip_cost,
+          driver_id: driver.id,
+          passenger_id: passenger.id
+        }
+      }
+
+      expect {
+        post passenger_trips_path(passenger.id), params: trip_hash
+      }.must_differ "Trip.count", 1
+
+      must_redirect_to passenger_path(passenger.id)
+
+    end 
```

Comment:
or create and update: would love to see that sweet negative case, in case trip creation/update wasn't successful (maybe sent without all the data). This would require trips to have validations, which it actually automatically does because of its relationships: it must have a passenger and driver! (y'all have tests to prove this ;) )